src/flat.js

```diff
@@ -0,0 +1,13 @@
+export function flat(depth = 1) {
+	return this.reduce(
```

Comment:
When binding is used, will we have the same issue here with `this.reduce` as we did with `this.map`?

Reply:
I'm not sure...but it doesn't hurt to use `Array.prototype.reduce.apply` as well.

Reply:
Done